src/en/hiveworks/src/eu/kanade/tachiyomi/extension/en/hiveworks/Hiveworks.kt

```diff
@@ -178,8 +178,27 @@ class Hiveworks : ParsedHttpSource() {
     // Included to call custom error codes
     override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
         return if (manga.status != SManga.LICENSED) {
+            val uri = Uri.parse(manga.url).buildUpon()
             client.newCall(chapterListRequest(manga))
-                .asObservableSuccess()
+                .asObservable()
+                .doOnNext { response ->
+                    if (!response.isSuccessful) {
```

Comment:
Should implement this check in an interceptor. See:
https://github.com/keiyoushi/extensions-source/blob/d7d84b3419b6c6e0fca2420957319323eecc41c4/src/en/xoxocomics/src/eu/kanade/tachiyomi/extension/en/xoxocomics/XoxoComics.kt#L27-L42

Sites tend to error on multiple pages, not just one.

Reply:
Updated. Thank you for this pointer. I did not find this pattern, although I would be happy to start writing an extension code style guide / common patterns list for the documentation, if that would be helpful.